src/order/disjointed.lean

```diff
@@ -8,83 +8,141 @@ import order.partial_sups
+/-- `disjointed f` is the unique sequence that is pairwise disjoint and has the same partial sups
+as `f`. -/
+lemma disjointed_unique {f d : ℕ → α} (hdisj : pairwise (disjoint on d))
+  (hsups : partial_sups d = partial_sups f) : d = disjointed f :=
 begin
-  rw disjointed,
-  generalize_hyp : f n = t at h₁ ⊢,
-  induction n,
-  case nat.zero { simp [nat.not_lt_zero, h₁] },
-  case nat.succ : n ih {
-    rw [Inter_lt_succ, inter_comm ((f n)ᶜ), ← inter_assoc],
-    exact h₂ _ n ih }
+  ext n,
+  cases n,
+  { rw [←partial_sups_zero d, hsups, partial_sups_zero, disjointed_zero] },
+  suffices h : d n.succ = partial_sups d n.succ \ partial_sups d n,
+  { rw [h, hsups, partial_sups_succ, disjointed_succ, sup_sdiff, sdiff_self, bot_sup_eq] },
+  rw [partial_sups_succ, sup_sdiff, sdiff_self, bot_sup_eq, eq_comm, sdiff_eq_self_iff_disjoint],
+  suffices h : ∀ m ≤ n, disjoint (partial_sups d m) (d n.succ),
+  { exact h n le_rfl },
+  rintro m hm,
+  induction m with m ih,
+  { exact hdisj _ _ (nat.succ_ne_zero _).symm },
+  rw [partial_sups_succ, disjoint_iff, inf_sup_right, sup_eq_bot_iff, ←disjoint_iff, ←disjoint_iff],
+  exact ⟨ih (nat.le_of_succ_le hm), hdisj _ _ (nat.lt_succ_of_le hm).ne⟩,
 end
 
-lemma disjointed_of_mono (hf : monotone f) :
-  disjointed f (n + 1) = f (n + 1) \ f n :=
-have (⋂ i (h : i < n + 1), (f i)ᶜ) = (f n)ᶜ,
-  from le_antisymm
-    (infi_le_of_le n $ infi_le_of_le (nat.lt_succ_self _) $ subset.refl _)
-    (le_infi $ λ i, le_infi $ λ hi, compl_le_compl $ hf $ nat.le_of_succ_le_succ hi),
-by simp [disjointed, this, diff_eq]
-
-open_locale classical
-
-lemma subset_Union_disjointed : f n ⊆ ⋃ i < n.succ, disjointed f i :=
-λ x hx,
-  have ∃ k, x ∈ f k, from ⟨n, hx⟩,
-  have hn : ∀ (i : ℕ), i < nat.find this → x ∉ f i,
-    from λ i, nat.find_min this,
-  have hlt : nat.find this < n.succ,
-    from (nat.find_min' this hx).trans_lt n.lt_succ_self,
-  mem_bUnion hlt ⟨nat.find_spec this, mem_bInter hn⟩
-
-lemma Union_disjointed : (⋃ n, disjointed f n) = (⋃ n, f n) :=
-subset.antisymm (Union_subset_Union $ λ i, inter_subset_left _ _)
-  (Union_subset $ λ n, subset.trans subset_Union_disjointed (bUnion_subset_Union _ _))
-
-lemma Union_disjointed_of_mono (hf : monotone f) (n : ℕ) :
-  (⋃ i < n.succ, disjointed f i) = f n :=
-subset.antisymm (bUnion_subset $ λ k hk, subset.trans disjointed_subset $ hf $ nat.lt_succ_iff.1 hk)
-  subset_Union_disjointed
-
-end set
+end generalized_boolean_algebra
+
+section complete_boolean_algebra
+variables [complete_boolean_algebra α]
+
+lemma le_supr_disjointed (f : ℕ → α) (n : ℕ) : f n ≤ ⨆ i ≤ n, disjointed f i :=
+by { rw [←partial_sups_eq_supr, partial_sups_disjointed], exact le_partial_sups _ _ }
+
+lemma supr_disjointed (f : ℕ → α) : (⨆ n, disjointed f n) = (⨆ n, f n) :=
+(supr_le_supr (λ n, disjointed_le f n)).antisymm
+  (supr_le (λ i, (le_supr_disjointed _ _).trans (bsupr_le_supr _ _)))
+
+lemma disjointed_eq_inf_compl (f : ℕ → α) (n : ℕ) :
+  disjointed f n = f n ⊓ (⨅ i < n, (f i)ᶜ) :=
```

Comment:
I think we should copy this one for set notation too, just for completeness. The proof is one line, right?